Assets/Scripts/PlayerControls.cs

```diff
@@ -148,11 +178,55 @@ void GroundMovement()
 		}
 	}
 
-	Vector3 GetInput()
+	/**
+	 * Raycast forward from the camera's position to detect
+	 * items to interact with.
+	 * 
+	 * This performs RaycastAll, and returns the closest collision.
+	 */
+	bool RaycastForward( out RaycastHit closestHit )
+	{
+		Vector3 camPos = Camera.main.transform.position;
+		Vector3 camForward = Camera.main.transform.forward;
+
+		Debug.DrawRay( camPos, camForward * _interactCheckDistance, Color.yellow, 1.0f, false );
+
+		RaycastHit[] hits = Physics.SphereCastAll(
+			new Ray( camPos, camForward ),
+			_interactCheckRadius,
+			_interactCheckDistance,
+			_cutting.cuttableLayer | _actor.actorResources.buddyLayer );
+
+		if ( hits.Length == 0 )
+		{
+			closestHit = new RaycastHit();
+			return false;
+		}
+
+		closestHit = hits[0];
+		float closestDistance = ( camPos - closestHit.point ).sqrMagnitude;
```

Comment:
It feels better to check this from the camera direction rather than where the player is facing. We should probably turn the player towards a buddy when giving them stuff.